### PR TITLE
test: cover SDK UTF-8 base64 helpers

### LINKED ISSUE #260: Add SDK unit tests for UTF-8 base64 helpers and runtime guards
## Summary

The TypeScript SDK uses dedicated base64 helpers to encode UTF-8 data and decode receipt headers in browser-compatible runtimes. Receipt tests cover common encoded JSON indirectly, but the helpers' Unicode, validation, and missing-runtime branches are not directly covered.

## Current behavior

The helpers:

- encode UTF-8 text through `TextEncoder` and `btoa`;
- decode base64 through `atob`, byte conversion, and `TextDecoder`;
- reject empty, padded incorrectly, whitespace-padded, or non-base64 input;
- return explicit errors when `atob` or `btoa` is unavailable.

## Desired behavior

Add direct, table-driven tests that lock down valid ASCII and Unicode round trips, strict invalid-input handling, and runtime capability errors. Production code should not change unless a very small testability refactor is required.

## Key interfaces

- `encodeUtf8ToBase64()` — ASCII, multibyte Unicode, emoji, and empty-text encoding behavior.
- `decodeBase64ToUtf8()` — valid decoding and strict base64 validation.
- Global `atob` and `btoa` capabilities — deterministic unavailable-runtime errors and restoration after each test.

## Acceptance criteria

- [ ] Tests round-trip plain ASCII, accented text, non-Latin text, and emoji.
- [ ] Tests compare at least one encoded result against a known base64 fixture.
- [ ] Decoding rejects an empty string, leading or trailing whitespace, invalid characters, and invalid padding.
- [ ] Missing `atob` produces the documented decoding-unavailable error.
- [ ] Missing `btoa` produces the documented encoding-unavailable error.
- [ ] Tests restore modified globals even when an assertion fails, so other SDK tests are isolated.
- [ ] Tests use Bun and do not depend on Node `Buffer` for the behavior under test.
- [ ] `cd sdk/typescript && bun run test && bun run typecheck` passes.

## Out of scope

- Changing the receipt schema or cryptographic verification.
- Adding URL-safe base64 support.
- Adding a third-party encoding dependency.
- Changing package publishing or build configuration.

## Blocked by

None — can start immediately.

### PR DESCRIPTION
**Pull request by saurabhhhcodes**
## Summary
- Add direct Bun tests for the TypeScript SDK UTF-8 Base64 helpers.
- Cover ASCII, accented, non-Latin, emoji, empty-text encoding, known fixtures, strict invalid-input rejection, and missing atob/btoa runtime errors.
- Restore overridden globals after capability tests.

## Validation
- cd sdk/typescript && bun run test -- src/__tests__/base64.test.ts
- cd sdk/typescript && bun run typecheck
- git diff --check

Fixes #260

> [!NOTE]
> ### Add tests for SDK UTF-8/base64 helper functions
> Adds [base64.test.ts](https://github.com/AnkanMisra/MicroAI-Paygate/pull/307/files#diff-73803b1f1547ad43974865d2324d9d73d06866cb401349e59aee16444af965cf) to cover `encodeUtf8ToBase64` and `decodeBase64ToUtf8`. Tests include round-trip encoding for ASCII, accented, non-Latin, and emoji strings, a known fixture check, rejection of invalid base64 with a specific error message, and behavior when `atob`/`btoa` globals are unavailable. A `withoutGlobal` helper temporarily removes and restores named globals during tests.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 2aff74b.</sup>
> 
>

## Summary by CodeRabbit

* **Tests**
  * Added coverage for UTF-8 and Base64 encoding and decoding, including Unicode text, empty input, and known fixtures.
  * Added validation for invalid Base64 data and environments without native Base64 support.
  * Confirmed runtime capabilities are restored after testing.

### CODE REVIEW

**Comment by AnkanMisra** [comment]
@codex review

File: sdk/typescript/src/__tests__/base64.test.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    }
+  }
+}
+
+describe("UTF-8 base64 helpers", () => {
+  it.each([
+    ["ASCII", "Hello, world!"],
+    ["accented text", "Café déjà vu"],
+    ["non-Latin text", "नमस्ते мир"],
+    ["emoji", "Paygate 🚀✅"],
```
Comment (AnkanMisra) [inline sdk/typescript/src/__tests__/base64.test.ts:30]: No emoji's

**Review by AnkanMisra — CHANGES_REQUESTED** [review]
